src/explorer/Common/ExploreMetric.cs

```diff
@@ -11,5 +11,11 @@ public interface ExploreMetric
 
         [JsonIgnore]
         public int Priority { get; }
+
+        public static TypedMetric<T> Create<T>(MetricDefinition<T> definition, T metric, int priority = 0)
+        where T : notnull
+        {
+            return new TypedMetric<T>(definition, metric, priority);
+        }
```

Comment:
I am wondering if maybe this `Create` method would be better in the `MetricDefinition` class:

```csharp
    public class MetricDefinition<T>
    where T : notnull
    {
        public MetricDefinition(string name) => Name = name;

        public string Name { get; }

        public ExploreMetric Create(T metric, int priority = 0)
        {
            return new TypedMetric<T>(this, metric, priority);
        }
    }
```

Then instead of this:
```csharp
yield return ExploreMetric.Create(MetricDefinitions.AverageEstimate, decimalResult);
```

We can write this:
```csharp
yield return MetricDefinitions.AverageEstimate.Create(decimalResult));
```

What do you think?

Reply:
I am not sure why I added the `ExploreMetric.Create` method. It should be usable to use the `TypedMetric` constructor and the code will look just like the original version.